### PR TITLE
Static build for Python

### PR DESCRIPTION
**Pull request by lanking520**
## Description ##
Add Python static build in ci for verification
@szha @zachgk @marcoabreu 

### Essentials ###
Please feel free to remove inapplicable items for your PR.
- [ ] The PR title starts with [MXNET-$JIRA_ID], where $JIRA_ID refers to the relevant [JIRA issue](https://issues.apache.org/jira/projects/MXNET/issues) created (except PRs with tiny changes)
- [ ] Changes are complete (i.e. I finished coding on this PR)
- [ ] All changes have test coverage:
- Unit tests are added for small changes to verify correctness (e.g. adding a new operator)
- Nightly tests are added for complicated/long-running ones (e.g. changing distributed kvstore)
- Build tests will be added for build configuration changes (e.g. adding a new build option with NCCL)
- [ ] Code is well-documented: 
- For user-facing API changes, API doc string has been updated. 
- For new C++ functions in header files, their functionalities and arguments are documented. 
- For new examples, README.md is added to explain the what the example does, the source of the dataset, expected performance on test set and reference to the original paper if applicable
- Check the API doc at http://mxnet-ci-doc.s3-accelerate.dualstack.amazonaws.com/PR-$PR_ID/$BUILD_ID/index.html
- [ ] To my best knowledge, examples are either not affected by this change, or have been fixed to be compatible with this change

### CODE REVIEW

File: tools/staticbuild/build_wheel.sh [hunk 1]
```diff
@@ -26,4 +26,5 @@ python setup.py bdist_wheel
 
 wheel_name=$(ls -t dist | head -n 1)
 pip install -U --user --force-reinstall dist/$wheel_name
-python sanity_test.py
+cd ..
+python nosetests -v $(ls tests/python/unittest/test_gluon*.py | grep -v data | grep -v model_zoo | grep -v utils | grep -v rnn)
```
Comment (marcoabreu) [inline tools/staticbuild/build_wheel.sh]: Could we run tests separately? When compiling, I wouldn't expect to run tests. Also, the build environments could might not be prepared to run mxnet (our GPU complication environment, for example, doesn't).

**Comment by larroy** [comment]
I might have missed something, but why are we using ubuntu 14.04? Latest LTS is 18.04...

**Comment by stu1130** [comment]
@mxnet-label-bot add [pr-awaiting-review]
Great work! @lanking520

**Comment by lanking520** [comment]
> I might have missed something, but why are we using ubuntu 14.04? Latest LTS is 18.04...

For publishing, we are still using 14.04 due to the `glibc` versioning problem.

**Comment by lanking520** [comment]
@szha @marcoabreu Please check again to see if this meet yours requirement

**Comment by marcoabreu** [comment]
Looks great, thanks a ton!

File: ci/publish/python/build.sh [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+set -ex
+
+source tools/staticbuild/build.sh $mxnet_variant pip
```
Comment (larroy) [inline ci/publish/python/build.sh]: why source?

**Pull request merged**